index.js

```diff
@@ -0,0 +1,236 @@
+
+const getRelevantURL = (url) => `${url}?fields=${RELEVANT_FIELDS}`;
+
+const freeChildren = (
+  container = document.getElementById("countries-container")
+) => {
+  while (container.firstChild) {
+    container.removeChild(container.firstChild);
+  }
+};
+
+const addInfoLi = (infoType, infoValue) => {
+  const li = document.createElement("li");
+  const strong = document.createElement("strong");
+  strong.innerText = `${infoType}: `;
+  li.appendChild(strong);
+  const p = document.createElement("p");
+  if (!infoValue) {
+    infoValue = "N/A";
+  }
+  p.innerText = infoValue.toString();
+  li.appendChild(p);
+  return li;
+};
+
+const addFlagDiv = (src) => {
+  const imageDiv = document.createElement("div");
+  imageDiv.className = "country-flag";
+  const flagImage = document.createElement("img");
+  flagImage.src = src;
+  imageDiv.appendChild(flagImage);
+  return imageDiv;
+};
+
+const addCountryInfo = (commonName, population, region, capital) => {
+  const countryInfoDiv = document.createElement("div");
+  countryInfoDiv.className = "country-info";
+  const h2 = document.createElement("h2");
+  h2.className = "country-title";
+  h2.innerHTML = commonName;
+  countryInfoDiv.appendChild(h2);
+
+  const ul = document.createElement("ul");
+  ul.className = "country-brief";
+  const population_li = addInfoLi("population", formatInteger(population));
+  const region_li = addInfoLi("region", region);
+  const capital_li = addInfoLi("capital", capital);
+  ul.appendChild(population_li);
+  ul.appendChild(region_li);
+  ul.appendChild(capital_li);
+  countryInfoDiv.appendChild(ul);
+  return countryInfoDiv;
+};
+
+const arrayToString = arr => {
+  let strResult = "";
+  if(arr && arr.length){
+    strResult = arr.join(',');
+  }
+  return strResult;
+}
+
+const buildQueryStr = queryData => {
+  const {commonName,capital,population,region,svg,nativeName,subregion,TLD,
+  borders,currencies,languages} = queryData;
+  const hrefP1 = `./details.html?name=${commonName}&capital=${capital}&population=${population}&region=${region}&flag=${svg}`;
+  const hrefP2 = `&nativeName=${nativeName}&subregion=${subregion}&TLD=${TLD}`;
+  const borderStr = arrayToString(borders);
+  const currecnyStr = arrayToString(Object.keys(currencies).map(key=>currencies[key].name));
+  const langStr = arrayToString(Object.keys(languages).map(key=>languages[key]));
+  return`${hrefP1}&${hrefP2}&borders=${borderStr}&currencies=${currecnyStr}&languages=${langStr}`;
+
+}
+
+const addCountryHTML = (countryData) => {
+  const {
+    flags,
+    capital,
+    name,
+    population,
+    region,
+    subregion,
+    currencies,
+    languages,
+    borders,
+    tld,
+  } = countryData;
+
+  const commonName = name.common;
+
+  let nativeName = name.nativeName;
+  const firstKey = Object.keys(nativeName)[0];
+  nativeName = firstKey && nativeName[firstKey] ? nativeName[firstKey].common : commonName;
```

Comment:
Instead of manipulating the `nativeName` variable, create a new one called countryDisplayName which will use `nativeName[firstKey].common` `commonName`